yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -552,8 +558,47 @@ private[spark] class ApplicationMaster(
     }
   }
 
+  /** set  "MapReduceShuffleToken" before registerAM
+    * @return
+    */
+  private def initJobCredentialsAndUGI() = {
+    val sc = sparkContextRef.get()
+    val conf = if (sc != null) sc.getConf else sparkConf
+    val isEncryptedShuffle = if (conf != null) {
+      conf.getBoolean("spark.encrypted.shuffle", false)
+    } else {
+      false
+    }
+    if (isEncryptedShuffle) {
```

Comment:
Another instance of this code that could use some cleaning up.

I wonder if it would make sense to have a `CryptoConf` object that wraps a `SparkConf` (like `SSLOptions`).